src/norm/postgres.nim

```diff
@@ -98,6 +98,14 @@ proc createTables*[T: Model](dbConn; obj: T) =
       fkGroups.add "FOREIGN KEY($#) REFERENCES $#($#)" %
         [obj.col(fld), typeof(get val.model).table, typeof(get val.model).col("id")]
 
+    when obj.dot(fld).hasCustomPragma(fk):
+      const pragVal = $(obj.dot(fld).getCustomPragmaVal(fk))
+      const fkTargetIsModel = isModel(none(typeof(obj.dot(fld).getCustomPragmaVal(fk))))
+      when val is SomeInteger and fkTargetIsModel:
+        fkGroups.add "FOREIGN KEY ($#) REFERENCES $#(id)" % [fld, $pragVal]
+      else:
+        {.error: "Pragma fk must be used on SomeInteger field and target a Model." .}
```

Comment:
Shouldn't that be a `raise newException`? Never seen this syntax before.

Reply:
Do you mean the error pragma ? An error output by the compiler seemd more approriate. I'll improve the message.

Reply:
@Clonkk Oh, is that the way to emit compile time errors? I thought you should just raise an exception in those cases, compile and runtime alike.

Reply:
Yes, it is. You can also use {.fatal: "..." .}. 
I think raising a compiler error is cleaner than an exception in this case considering it's a compile constraint.